### PR TITLE
Adds Flavor Text

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Adds flavor text to characters! You can have up to 512 characters, and is visible on-examine IF you are not marked as an unknown AND your `name` matches your `real_name`. For extra fun, you can |italicize|, +bold+, and _underline_ your flavor text.
Code partially lifted from Donkstation and rewritten to fix HTML encoding exploits and adding the aforementioned special text.

## Why It's Good For The Game
It gives more depth to characters, it feels sorta required for a living, breathing character.

## Changelog
:cl:
add: Flavor text has been added! Edit it in character setup.
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1349,7 +1359,17 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
 						to_chat(user, "<font color='red'>Invalid name. Your name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z, -, ' and .</font>")
-
+				//MONKESTATION EDIT START - FLAVOR TEXT
+				if("flavor_text")
+					var/new_text = input(user, "Please enter new flavor text (appears on examine):", "Character Preference") as text|null
+					if(!new_text)
+						return
+					if(length(new_text) > MAX_FLAVOR_LEN)
+						alert("Your flavor text is too long. It must be no more than [MAX_FLAVOR_LEN] characters long. The current text will be trimmed down to meet the limit.")
+						new_text = copytext(new_text, 1, MAX_FLAVOR_LEN+1)
```
Comment (nednaZ) [inline code/modules/client/preferences.dm:1369]: Why is it MAX_FLAVOR_LEN+1?
The Bee coders were trying to figure this out as well when they were looking into porting the Donk version.

Comment (Zonespace27) [reply code/modules/client/preferences.dm:1369]: as far as I know, copytext copies from the start (1) up to the max (MAX_FLAVOR_LEN+1), will test and update code if needed

Comment (Zonespace27) [reply code/modules/client/preferences.dm:1369]: Not an issue, as per byond docs:
image

File: code/modules/mob/living/carbon/human/examine.dm [hunk 2]
```diff
@@ -16,6 +16,13 @@
 
 	. = list("<span class='info'>*---------*\nThis is <EM>[!obscure_name ? name : "Unknown"]</EM>!")
 
+	//MONKESTATION EDIT START - FLAVOR TEXT
+	if(flavor_text && !obscure_name)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/examine.dm:20]:
```suggestion
	if(flavor_text && !obscure_name && (real_name == name))
```

**Comment by nednaZ** [comment]
Works for me.
Different IDs and face coverings all prevent flavor text from being shown, so disguises should still function.

Only request is for the input box to have some kind of reference for the text modifiers.

**Comment by Zonespace27** [comment]
> Works for me. Different IDs and face coverings all prevent flavor text from being shown, so disguises should still function.
> 
> Only request is for the input box to have some kind of reference for the text modifiers.

Done

**Review by nednaZ — APPROVED** [review]
This looks like the best method we can get for flavor text without having too many metagamer issues.
Approved for merge from me.

**Pull request merged**